### PR TITLE
Update dg for model and storage

### LINKED ISSUE #150: Model API + Storage API Component DG update

### PR DESCRIPTION
**Pull request by bobbylaw**
- [x] DG and diagrams for model
- [x] DG and diagrams for Storage
- [x] Find Implementation DG

Found a bug on findcommand
Findcommand able to take in two prefix, which should not be the case

- [x] fix findcommand bug

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -96,26 +96,52 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
...
+
+`Model`
+
+* stores a `UserPref` object that represents the users preference
+
+`InventoryModelManager`
+
+* stores a comparator used to sort the filtered list
+* stores the inventory book data
+* stores a list of InventoryBook
```
Comment (Choo-Xing-Yu) [inline docs/DeveloperGuide.md]: maybe state what for!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -96,26 +96,52 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
...
+* exposes an unmodifiable `ObservableList<Item>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+
+`DeliveryModelManager`
+
+* stores the delivery book data
+* stores a list of DeliveryBook
```
Comment (Choo-Xing-Yu) [inline docs/DeveloperGuide.md]: same here

**Comment by Choo-Xing-Yu** [comment]
image
There's a couple of errors in the diagram! Check out the ss i tagged

**Comment by Choo-Xing-Yu** [comment]
image
And another update! I dont see a uniquetaglist class in the folder unless im mistaken! and i dont think we should show inventory book in the item class diagram

**Comment by Choo-Xing-Yu** [comment]
image
Remember multiplicity and deliverybookstorage is an interface!

**Comment by bobbylaw** [comment]
@xnoobftw my bad for overlooking the interfaces, I have edited the diagrams and added my find implementation

**Comment by Choo-Xing-Yu** [comment]
image
Is there anyway to adjust so that the multiplicity is not blocked? Try increasing the lenght of the arrow

**Comment by Wincenttjoi** [comment]
> image
> There's a couple of errors in the diagram! Check out the ss i tagged

Does inventoryModelManager has Item? Not sure about this, or is it referring to the `FilteredList<Item> filtered Items`?
Same goes for deliveryModelManager

**Comment by Wincenttjoi** [comment]
> image
> And another update! I dont see a uniquetaglist class in the folder unless im mistaken! and i dont think we should show inventory book in the item class diagram

Actually I think its fine with or without the InventoryBook, because the reader can see how its connected to the component outside Item. Seems like @zeranium97 has changed it to without but im fine with either! (Personally think with is better tho)

**Comment by bobbylaw** [comment]
> > image
> > There's a couple of errors in the diagram! Check out the ss i tagged
> 
> Does inventoryModelManager has Item? Not sure about this, or is it referring to the `FilteredList<Item> filtered Items`?
> Same goes for deliveryModelManager

Yes it is referring to the filteredlist<Items>

**Comment by Wincenttjoi** [comment]
LGTM Thanks!

**Comment by Wincenttjoi** [comment]
@zeranium97 realized that ur  structure of items object and delivery object are not explained at all. U want to add the explanation urself or I do it for you? Also its a class diagram, not object diagram

**Pull request merged**